xrpl/models/utils.py

```diff
@@ -1,10 +1,73 @@
 """Helper util functions for the models module."""
 
+import re
 from dataclasses import dataclass, is_dataclass
-from typing import Any, Dict, List, Type, TypeVar, cast
+from typing import Any, Dict, List, Optional, Pattern, Type, TypeVar, cast
+
+from typing_extensions import Final
 
 from xrpl.models.exceptions import XRPLModelException
 
+HEX_REGEX: Final[Pattern[str]] = re.compile("[a-fA-F0-9]*")
+
+# Credentials are represented in hex. Whilst they are allowed a maximum length of 64
+# bytes, every byte requires 2 hex characters for representation
+_MAX_CREDENTIAL_LENGTH: Final[int] = 64 * 2
+
+MAX_CREDENTIAL_ARRAY_LENGTH = 8
+
+
+def get_credential_type_error(credential_type: str) -> Optional[str]:
+    """
+    Utility function for validating the CredentialType field in all
+    Credential-related transactions.
+
+    Args:
+        credential_type: A (hex-encoded) value to identify the type of credential from
+            the issuer.
+
+    Returns:
+        Errors, if any, during validation of credential_type field
+    """
+    errors = []
+    # credential_type is a required field in this transaction
+    if len(credential_type) == 0:
+        errors.append("Length must be > 0.")
```

Comment:
```suggestion
        errors.append("Length cannot be an empty string.")
```

Reply:
If its alright with you, I slightly prefer the existing version of the error message. If I incorporate this change, I'll need to update the unit tests as well.

Besides, there have been several back-and-forths amongst the reviewers earlier in this PR.